src/mdast-process-images.js

```diff
@@ -12,29 +12,72 @@
 import { visit } from 'unist-util-visit';
 import processQueue from '@adobe/helix-shared-process-queue';
 
+function createFilter(log, baseUrlStr, imgSrcPolicy) {
+  const baseUrl = new URL(baseUrlStr);
+  const imgSrcPolicFilters = imgSrcPolicy.map((policyValue) => {
+    // filter according to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/Sources
+    if (policyValue === '*') {
+      return () => true;
+    }
+
+    if (policyValue === 'self') {
+      return (url) => baseUrl.host === url.host;
+    }
+
+    return (url) => {
+      // omit the protocol check, as https:// is enforced earlier already
+      // omit the pathname check, as we may not need it (yet)
+      const protocolEndIdx = policyValue.indexOf('://');
+      let host = protocolEndIdx < 0 ? policyValue : policyValue.substring(protocolEndIdx + 3);
+      const slashIdx = host.indexOf('/');
+      if (slashIdx >= 0) host = host.substring(0, slashIdx);
```

Comment:
```suggestion
      if (slashIdx >= 0) {
        host = host.substring(0, slashIdx);
      }
```

this should be detected by eslint... we don't allow single line conditions